nix_update/update.py

```diff
@@ -8,21 +8,26 @@
 from .options import Options
 from .utils import info, run
 from .version import fetch_latest_version
-from .version.version import VersionPreference
+from .version.version import Version, VersionPreference
 from .git import old_version_from_git
 
 
 def replace_version(package: Package) -> bool:
+    assert package.new_version is not None
     old_version = package.old_version
-    new_version = package.new_version
-    assert new_version is not None
```

Comment:
We need to run the check before `package.new_version.number` because `package.new_version` might be None. Before the assignments seems cleaner than in between the lines.